### PR TITLE
Remove otp 20 from MIM

### PR DESCRIPTION
**Pull request by aleklisi**
This PR removes support for OTP 20.* and older from CircleCI and older OTPs deprecation message.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7318.1](https://travis-ci.org/esl/MongooseIM/jobs/598671275) / Erlang 22.0 / small_tests / 0dee10824024e0947f9ccad3d8012ae3da4c9812
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2495/7318/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2495/7318/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-16_15.31.13/lib.mongooseim.logs/run.2019-10-16_15.31.13/suite.log.html)

File: rebar.config [hunk 1]
```diff
@@ -22,8 +22,8 @@
   {".*", "priv/lib/mongoose_mam_id.so", ["c_src/mongoose_mam_id.cpp"], [{env, [{"CXXFLAGS", "$CXXFLAGS -std=c++11"}]}]}
  ]}.
 
-%% We use functions introduced in Erlang/OTP 19
-{require_min_otp_vsn, "20"}.
+%% We use syntax introduced in Erlang/OTP 21
```
Comment (NelsonVides) [inline rebar.config]: We (will) use both functions and syntax introduced in Erlang/OTP 21, like logger or persistent_term, is just that for now we have only updated the stacktrace thing.

Comment (aleklisi) [reply rebar.config]: hmm, I would rather remove the comment completely instead of manainig it as the option is rather self-explanatory :)

Comment (NelsonVides) [reply rebar.config]: Great as well. Btw, you could have pushed with [skip ci] or so, now you just spend a dollar on CircleCI :stuck_out_tongue:

**Pull request merged**